code/modules/power/supermatter/supermatter.dm

```diff
@@ -630,6 +642,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			power_transmission_bonus += gas_comp[gasID] * gas_trans[gasID] * (isnull(transit_mod[gasID]) ? 1 : transit_mod[gasID])
 		power_transmission_bonus *= h2obonus
 
+		//Miasma is really just microscopic particulate. It gets consumed like anything else that touches the crystal.
+		if(gas_comp[/datum/gas/miasma])
+			var/miasma_pp = env.return_pressure() * gas_comp[/datum/gas/miasma]
+			var/consumed_miasma = clamp(((miasma_pp - MIASMA_CONSUMPTION_PP) / (miasma_pp + MIASMA_PRESSURE_SCALING)) + gasmix_power_ratio, MIASMA_CONSUMPTION_RATIO_MIN, MIASMA_CONSUMPTION_RATIO_MAX) * gas_comp[/datum/gas/miasma] * combined_gas
+			if(consumed_miasma)
+				removed.gases[/datum/gas/miasma][MOLES] -= consumed_miasma
+				matter_power += consumed_miasma * MIASMA_POWER_GAIN
```

Comment:
How much has this been tested?

Reply:
Not much:
I basically just loaded metastation, setup the SM as default, and then dumped two full canisters of miasma into the chamber.
The supermatter reached around the same power level as it would with emitters and slowly dropped over several minutes.